### PR TITLE
[NPU] fix dtype error in sampling backend ascend

### PR DESCRIPTION
**Pull request by Makcum888e**
## Motivation

`torch_npu.npu_top_k_top_p` requires same `dtype` for `logits` and `top_p`. Currently for models like CosyVoice3 from SGLang-Omni it produced error as `logits` come in `blfloat16` while `top_p` in `float32`.

## Modifications

`top_p` casted to `logits.dtype`. Make calculation in original dtype

## Speed Tests and Profiling

Enabling `--tts_engine.engine.sampling-backend ascend` gives `5%` speed up for CosyVoice3

<details>
    <summary>results</summary>

sampling backend pytorch (before)
```
sgl-omni serve --model-path /home/weights/FunAudioLLM/Fun-CosyVoice3-0.5B-2512/
```
```
============================================================
                   Speed Benchmark Result                   
============================================================
  Model:                         fishaudio/s2-pro
  Concurrency:                   10
  Completed requests:            1088
  Failed requests:               0
------------------------------------------------------------
  Latency mean (s):              5.269
  Latency median (s):            4.909
  Latency p95 (s):               7.018
  Latency p99 (s):               8.83
  RTF mean:                      1.1238
  RTF median:                    1.055
  RTF p95:                       1.6849
  RTF p99:                       2.345
  Audio duration mean (s):       5.141
  Audio throughput (s/s):        9.746
  Output throughput (tok/s):     243.7
  Output tokens/request-s:       48.2
  Output tokens (mean):          129
  Output tokens (total):         139848
  Prompt tokens (mean):          154
  Prompt tokens (total):         167806
  Throughput (req/s):            1.896
============================================================
``` 

sampling backend ascend (after)
```
sgl-omni serve --model-path /home/weights/FunAudioLLM/Fun-CosyVoice3-0.5B-2512/ --tts_engine.engine.sampling_backend ascend
```
```
============================================================
                   Speed Benchmark Result                   
============================================================
  Model:                         fishaudio/s2-pro
  Concurrency:                   10
  Completed requests:            1088
  Failed requests:               0
------------------------------------------------------------
  Latency mean (s):              4.735
  Latency median (s):            4.271
  Latency p95 (s):               6.456
  Latency p99 (s):               7.712
  RTF mean:                      1.0174
  RTF median:                    0.9537
  RTF p95:                       1.6132
  RTF p99:                       1.9351
  Audio duration mean (s):       5.125
  Audio throughput (s/s):        10.759
  Output throughput (tok/s):     269.0
  Output tokens/request-s:       58.7
  Output tokens (mean):          128
  Output tokens (total):         139389
  Prompt tokens (mean):          154
  Prompt tokens (total):         167806
  Throughput (req/s):            2.099
============================================================
```

</details>

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #35732786234](https://github.com/sgl-project/sglang/actions/runs/35732786234)
Latest PR Test (Extra): :x: [Run #35732785716](https://github.com/sgl-project/sglang/actions/runs/35732785716)
Latest PR Test (AMD ROCm 10): :x: [Run #35732786340](https://github.com/sgl-project/sglang/actions/runs/35732786340)

### CODE REVIEW

File: python/sglang/srt/layers/sampler.py [hunk 1]
```diff
@@ -791,6 +791,7 @@ def top_k_top_p_min_p_sampling_from_logits_ascend(
     if hasattr(torch_npu, "npu_top_k_top_p") and torch.all(
         (top_ks <= 1024) & (top_ks >= 1)
     ):
+        top_ps = top_ps.to(logits.dtype)
```
Comment (ping1jing2) [inline python/sglang/srt/layers/sampler.py:794]: could you recheck `(top_ks <= 1024) & (top_ks >= 1)` this line to confirm if we should change it like https://github.com/sgl-project/sglang/pull/39404

Comment (Makcum888e) [reply python/sglang/srt/layers/sampler.py:794]: this PR gives good improvement in Sampler in general, but it seems that for our benchmark case we have slow down 5% due to less optimal batching.

Comment (Makcum888e) [reply python/sglang/srt/layers/sampler.py:795]: root cause was a compilation time increased as more different batch sizes comes in prefill time, with warmup for all batch sizes this optimization works too
